scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1519,10 +1530,17 @@
                      ;; TODO: Rescind the offer in fenzo
                      )
     (framework-message [this driver executor-id slave-id message]
-                       ;; the executor-id is the same as the task-id (see cook.mesos.task/task-info->mesos-message)
-                       (async-in-order-processing (:value executor-id)
-                                                  #(handle-framework-message conn handle-progress-message message))
-                       (heartbeat/notify-heartbeat heartbeat-ch executor-id slave-id message))
+                       (try
+                         (let [{:strs [task-id] :as parsed-message} (json/read-str (String. ^bytes message "UTF-8"))
+                               executor (task-id->executor (d/db conn) task-id)]
```

Comment:
We query the database without using the result when the message is a Cook Executor message.

Reply:
Isn't the result from the database being used to avoid forwarding the message to the `heartbeat` mechanism.